src_lockfree/michael_scott_queue.ml

```diff
@@ -34,31 +34,31 @@ let is_empty { head; _ } = Atomic.get (Atomic.get head) == Nil
 exception Empty
 
 let pop_opt { head; _ } =
-  let b = Backoff.create () in
-  let rec loop () =
+  let b = Backoff.default in
+  let rec loop b =
     let old_head = Atomic.get head in
     match Atomic.get old_head with
     | Nil -> None
     | Next (value, next) when Atomic.compare_and_set head old_head next ->
         Some value
     | _ ->
-        Backoff.once b;
-        loop ()
+        let b = Backoff.once b in
+        loop b
   in
-  loop ()
+  loop b
```

Comment:
I'd probably drop the `let b = Backoff.default` and write `loop Backoff.default` here, but this is mostly stylistic.